payment/src/test/java/org/killbill/billing/payment/api/TestPaymentApi.java

```diff
@@ -147,12 +147,43 @@ public void testAddRemovePaymentMethod() throws Exception {
 
         checkPaymentMethodPagination(paymentMethodId, baseNbRecords + 1, false);
 
-        paymentApi.deletePaymentMethod(account, paymentMethodId, true, ImmutableList.<PluginProperty>of(), callContext);
+        paymentApi.deletePaymentMethod(account, paymentMethodId, true, false, ImmutableList.<PluginProperty>of(), callContext);
 
         checkPaymentMethodPagination(paymentMethodId, baseNbRecords, true);
     }
 
     @Test(groups = "slow")
+    public void testAddRemovePaymentMethodWithForcedDeletion() throws Exception {
+        final Long baseNbRecords = paymentApi.getPaymentMethods(0L, 1000L, false, ImmutableList.<PluginProperty>of(), callContext).getMaxNbRecords();
+        Assert.assertEquals(baseNbRecords, (Long) 1L);
+
+        final Account account = testHelper.createTestAccount(UUID.randomUUID().toString(), true);
+        final UUID paymentMethodId = account.getPaymentMethodId();
+
+        checkPaymentMethodPagination(paymentMethodId, baseNbRecords + 1, false);
+
+        paymentApi.deletePaymentMethod(account, paymentMethodId, false, true, ImmutableList.<PluginProperty>of(), callContext);
+
```

Comment:
@sbrossie 
I've added (injected) the TagUserApi but when I try to get all the Tags associated with the Account, the list is empty.
I was debugging the test, and I have found out that when the tag is being "added" inside the [setAccountAutoPayOff](https://github.com/javier-gomez-hs/killbill/blob/60618ff139816a46465f01feabdfb4c769dfc276/payment/src/main/java/org/killbill/billing/payment/core/ProcessorBase.java#L103-L103) method, the **tagInternalApi** used is a Mock.
I wasn't able to find where is it being set as a Mock instead of the "real" **tagInternalApi**.
Any thoughts?

Thanks!